src/server/handlers/launchpad.js

```diff
@@ -672,19 +672,32 @@ export const getSnapBuilds = async (req, res) => {
     const snap = await getLaunchpad().get(snapUrl);
     const builds = await internalGetSnapBuilds(snap, req.query.start, req.query.size);
 
+    // Let's collect corresponding build annotations.
+    let build_annotations = {};
```

Comment:
Ideally it would be great for collecting annotations to be part of `internalGetSnapBuilds`, so that `internalGetSnapBuilds` already returns annotated builds. In current implementation annotations are returned separately to the UI, which means that it's the front-end that will need to join annotations with build object.
Preferably this join would be done in `internapGetSnapBuilds`, so that any part of the app that uses `internalSnapBuilds` would get annotated builds.

If that seems unnecessary, I'd suggest at least extracting getting annotations to some `internalGetBuildAnnotations` method, so the logic is not hidden only in `getSnapBuilds` handler.

Reply:
I've isolated `internalGetBuildAnnotations` and `getBuildId(<build>)` common helper so the client work is clearer. I thought it's better to keep build records pristine, as they come from LP, for the time being.